### ISSUE TITLE
Add heads Gnu/Linux Libre

### ISSUE THREAD

**Issue by geplus** [nylira/prism-break#1931]
Hi,
from Distrowatch:
"heads is a privacy-focused Linux distribution designed to make it easy for users to access the Internet anonymously using the Tor network. heads is based on Devuan and features only free (libre) software. The Linux kernel has had non-free blobs removed." 
Here the official heads about page:
https://heads.dyne.org/about.html
You can simply mentioning heads in the notes of the recommended linux distros, if you like.

**Comment by yegortimoshenko** [comment]
I do not think that listing distros with deblobbed kernel rebranded by GNU would be a good idea. It segments community by creating downstream users without any particular technical advantages. If you have a machine that doesn't rely on kernel blobs, kernel won't load them anyway. If you don't want to install proprietary software, just don't install it.

Also, proprietary hardware is not any better from practical (freedom, privacy) perspective than proprietary firmware, both can be malicious and both are used to perpetuate obsolescence, restrict hardware manufacturers, and ship malicious features. Firmware doesn't magically become hardware when it's flashed on a ROM.

In context of Tails particularly, not shipping blobs just reduces value of having a live operating system bootable from a pen drive because networking won't work on most hardware without blobs.

**Comment by yegortimoshenko** [comment]
> In heads, the init of choice is not systemd. systemd is a huge piece of software that, while being free software, has not been audited for security since its creation. Being big as it is, it is hard to do so, and as time goes, it's becoming even tougher to audit systemd. We do not aim to disrespect or get into the controversy on why systemd is a bad choice. We just do not wish to use it.

Like heads authors, I'm not going to go into systemd debate, but from security perspective, systemd provides private `/tmp` in mount namespace, temporary users, socket activation can be used to prevent leaks, `systemd-boot` is much smaller and easier to audit than GRUB, etc.

**Comment by hasufell** [comment]
> I do not think that listing GNU-oriented distros with deblobbed kernel would be a good idea.

That is utter nonsense.

Prism-break is about **software** and not hardware. Having a deblobbed kernel **is** a plus, because it exposes hardware that doesn't run with **free software**.

Whether the hardware itsels if "free" or "open" is outside of prism-break scope.

The reason we haven't listed many distros yet that do deblobbing by default is that deblobbing is really just a shell script that can be run on any distro if the user chooses to build the kernel himself. However, if the kernel is already deblobbed, that's still a plus, but not a fundamental one.

**Comment by yegortimoshenko** [comment]
And what makes firmware software? You speak like there is a definitive line between the two, while there isn't any. Having firmware pre-flashed on the chip doesn't turn it from software into hardware.

**Comment by hasufell** [comment]
> And what makes firmware software?

:roll_eyes:

**Comment by yegortimoshenko** [comment]
The same chunk of Verilog logic can often be taped out in hardware or loaded as firmware. This is only a technicality. The biggest reason why firmware is proprietary is that hardware is proprietary, and the former is very intimately hardware-specific. If anything, auditing/reverse-engineering firmware is easier than a taped out chip, or chip with firmware flashed on a ROM (that can be usually reflashed by manufacturer but not by user).

In other words, pushing the problem outside of the realm of software doesn't resolve it, it actually makes it worse because now you can't audit firmware and say there is no guarantee that you haven't received a special edition of the pre-flashed firmware, no one can replace it with free firmware, etc.

**Comment by yegortimoshenko** [comment]
> The reason we haven't listed many distros yet that do deblobbing by default is that deblobbing is really just a shell script that can be run on any distro if the user chooses to build the kernel himself. However, if the kernel is already deblobbed, that's still a plus, but not a fundamental one.

I've seen you say that, but I must say this is a weak argument: no one is going to run a shell script and build every kernel update manually. Most people don't run Gentoo :-)

**Comment by hasufell** [comment]
> I've seen you say that, but I must say this is a weak argument: running shell script on every kernel release manually is very inconvenient.

First, that is nonsense: For people who build their own kernel this is a no-brainer of wget + executing the script.

Second, you are contradicting yourself. First you want to exclude distros that ship with pre-deblobbed kernel, then you argue deblobbing yourself is too inconvenient. Aha.

> Most people don't run Gentoo :-)

This is not about gentoo in any way. I don't know what you mean by that.

**Comment by yegortimoshenko** [comment]
> Second, you are contradicting yourself. First you want to exclude distros that ship with pre-deblobbed kernel, then you argue deblobbing yourself is too inconvenient. Aha.

I'm not, I do think deblobbed kernels are a pretty bad idea unless one only uses free hardware. It's just that "end user can deblob" argument is weak because no one will actually do that. It would be a recommendation comparable to sending people over to Linux From Scratch site.

> Also, I'm not engaging in the hardware vs software definition nonsense discussion, since that is just a weak troll angle in my view. Prism-break focuses on software and we have established that several times.

This is willful ignorance, at cost of reducing overall end-user freedom and privacy. PRISM Break's goal is to fight mass surveillance and help persons retain their privacy, and free software is means to that end and not the end goal in and of itself.

My points:

* pre-flashed firmware can't be audited (unlike kernel blobs)
* no one can write free firmware replacement (see OpenFWWF)
* most of the time ROM chips can be reflashed by manufacturer using proprietary reflashing process, but not by end user, placing manufacturer in a position where they can pre-flash practically undetectable backdoor
* there is no distinct line between hardware and software: the same Verilog code can be taped out, or compiled into firmware, or loaded into FPGA
* most (not all) of the currently available hardware that doesn't require blobs is worse from freedom/privacy perspective than counterparts that require blobs
* situation is less than ideal either way, looking forward to usable free hardware

**Comment by yegortimoshenko** [comment]
All that being said, Heads seems to have improvements over Tails: https://puri.sm/posts/demonstrating-tamper-detection-with-heads/
